Feature: Add Docker image build configuration to pom.xml

sidhantmourya opened the pull request:
Introduced a docker.image.name property and configured the Maven plugin to build a Docker image with the specified name. This streamlines the Docker image creation process as part of the Maven build lifecycle.
This Fixes Issue #32

sidhantmourya committed:
Feature: Add Docker image build configuration to pom.xml

Introduced a docker.image.name property and configured the Maven plugin to build a Docker image with the specified name. This streamlines the Docker image creation process as part of the Maven build lifecycle.

firasrg commented on pom.xml:
```diff
@@ -28,6 +28,7 @@
 		<url/>
 	</scm>
 	<properties>
+		<docker.image.name>autocare-rest-api:latest</docker.image.name>
```
We can avoid setting the name statically, by using some built-in variables in the pom.xml file : `${project.groupId}/${project.artifactId}:${project.version}`

sidhantmourya replied:
I was not aware that we could do it in such a way.

firasrg replied:
No problem my friend, we learn together ! It's a good practice to make the image name starts with the org name, currently we can consider the `groupId`, but probably it will change later when we move to DockerHub

firasrg commented on pom.xml:
```diff
@@ -102,7 +103,17 @@
 							<artifactId>lombok</artifactId>
 						</exclude>
 					</excludes>
+					<imageName>${docker.image.name}</imageName> <!-- You can change this -->
```
This is not a valid code to declare image name in the plugin, did you try this out locally ?!

sidhantmourya replied:
Yes, it worked, I had to pass the argument "-Ddocker.image.name=spring-app" once the images were built, docker images was showing the images with same name used in the argument.

firasrg replied:
The argument you used is correct in terms of syntax and usage, but that's **not** what we're asking for. We need to avoid passing that argument, and have a fixed config in the `pom.xml`.

firasrg commented on pom.xml:
```diff
@@ -102,7 +103,17 @@
+
 				</configuration>
+				<executions>
+					<execution>
```
There is no need to add this execution while in _dev mode_. It would be great to use [maven profiles ](https://maven.apache.org/guides/introduction/introduction-to-profiles.html) for this. for example if profile is NOT "dev", then activate this execution.

sidhantmourya replied:
I haven't worked with Maven much, and its profiles, so it's all new to me, still learning it.

firasrg replied:
No problem my friend, we learn together ! Maven is one of the biggest build tools in Java world. take your time to explore its abilities

firasrg reviewed (changes requested):
Hi @sidhantmourya !

Thank you for the contribution! I noticed that the Docker image name declaration in the `spring-boot-maven-plugin` is not valid. Did you try this out locally ?

sidhantmourya committed:
Update Docker image naming and add build profile

Docker image name now uses groupId, artifactId, and version for better traceability. The build-image goal execution was moved to a new 'docker-build' Maven profile to allow more flexible Docker image creation.

sidhantmourya commented:
@firasrg i have updated the code to use build profiles and image name generation internally without using any external args.

firasrg commented:
Hi @sidhantmourya ! Sounds great, please know that a new review is in progress.
